src/main/java/edu/unc/lib/boxc/migration/cdm/services/PermissionsService.java

```diff
@@ -216,6 +245,61 @@ private List<String> queryForMappedIds(PermissionMappingOptions options) {
         return mappedIds;
     }
 
+    private List<List<String>> updateCsvRecords(PermissionMappingOptions options, Path permissionsMappingPath) throws Exception {
+        List<List<String>> updatedRecords = new ArrayList<>();
+        List<String> cdmIds = new ArrayList<>();
+        String everyoneField = getAssignedRoleValue(options.isStaffOnly(), options.getEveryone());
+        String authenticatedField = getAssignedRoleValue(options.isStaffOnly(), options.getAuthenticated());
+
+        // update existing entry
+        for (CSVRecord record : previousStatePermissions) {
+            cdmIds.add(record.get(0));
+            if (record.get(0).contains(options.getCdmId())) {
+                updatedRecords.add(Arrays.asList(options.getCdmId(), everyoneField, authenticatedField));
+            } else {
+                updatedRecords.add(Arrays.asList(record.get(0), record.get(1), record.get(2)));
+            }
+        }
+
+        // add new entry
+        if (!cdmIds.contains(options.getCdmId())) {
+            updatedRecords.add(Arrays.asList(options.getCdmId(), everyoneField, authenticatedField));
+        }
+
+        return updatedRecords;
+    }
+
+    public void capturePreviousState() {
+        if (Files.exists(project.getPermissionsPath())) {
+            setPreviousStatePermissions(getPreviousStatePermissions());
+        }
+    }
+
+    public void setPreviousStatePermissions(List<CSVRecord> previousStatePermissions) {
+        this.previousStatePermissions = previousStatePermissions;
+    }
+
+    public List<CSVRecord> getPreviousStatePermissions() {
+        if (previousStatePermissions == null) {
+            List<CSVRecord> permissions = new ArrayList<>();
+
+            try (
+                    Reader reader = Files.newBufferedReader(project.getPermissionsPath());
+                    CSVParser csvParser = new CSVParser(reader, CSVFormat.DEFAULT
+                            .withFirstRecordAsHeader()
+                            .withHeader(PermissionsInfo.CSV_HEADERS)
+                            .withTrim());
+            ) {
+                permissions = csvParser.getRecords();
+            } catch (IOException e) {
+                log.error("Failed to list permissions", e);
+            }
+
+            setPreviousStatePermissions(permissions);
```

Comment:
Are we anticipating needing to hold onto the previous state for comparison in a summary report? Just wondering since as far as i can tell its only being used for one purpose at the moment, which is in `updateCsvRecords`, so I wasn't sure what the caching and prepopulating of the value was for in this case.

Reply:
No, I don't think we need to hold onto the previous state. I was just having trouble getting the previous permissions so I tried using the previous state. I figured out a simpler solution (call `updateCsvRecords` before writing to the permissions csv).